### DISCUSSION TITLE
cant make it run on raspberry pi zero w

### DISCUSSION THREAD

**Discussion by alexbell62** [mKeRix/room-assistant#345] [Q&A]
Hi,
Ive just read about room assistant on the shiny https://www.room-assistant.io/ website. Really excited about it!
So i bought 6 raspberry pi zero W as it is one of the suggested devices on the wbsite.

Today i spent hours trying to make it work with no luck so far. I wonder if i can get some help here, please?
I followed all the instructions [here](https://www.room-assistant.io/guide/quickstart-pi-zero-w.html) and [here](https://www.room-assistant.io/integrations/bluetooth-low-energy.html#requirements).

An all the devices i tried it i get the following error:

> 
> pi@bedroom:~/room-assistant $ room-assistant
> *** WARNING *** The program 'node' uses the Apple Bonjour compatibility layer of Avahi.
> *** WARNING *** Please fix your application to use the native API of Avahi!
> *** WARNING *** For more information see <http://0pointer.de/blog/projects/avahi-compat.html>
> *** WARNING *** The program 'node' called 'DNSServiceRegister()' which is not supported (or only supported partially) in the Apple Bonjour compatibility layer of Avahi.
> *** WARNING *** Please fix your application to use the native API of Avahi!
> *** WARNING *** For more information see <http://0pointer.de/blog/projects/avahi-compat.html>
> 11/13/2020, 5:52:51 PM - info - IntegrationsModule: Loading integrations: home-assistant, bluetooth-low-energy
> 11/13/2020, 5:52:56 PM - info - NestFactory: Starting Nest application...
> 11/13/2020, 5:52:56 PM - info - InstanceLoader: AppModule dependencies initialized
> 11/13/2020, 5:52:56 PM - info - InstanceLoader: ConfigModule dependencies initialized
> 11/13/2020, 5:52:56 PM - info - InstanceLoader: NestEmitterModule dependencies initialized
> 11/13/2020, 5:52:56 PM - info - InstanceLoader: IntegrationsModule dependencies initialized
> 11/13/2020, 5:52:56 PM - info - InstanceLoader: HttpModule dependencies initialized
> 11/13/2020, 5:52:57 PM - info - InstanceLoader: DiscoveryModule dependencies initialized
> 11/13/2020, 5:52:57 PM - info - InstanceLoader: ClusterModule dependencies initialized
> 11/13/2020, 5:52:57 PM - info - InstanceLoader: TerminusModule dependencies initialized
> 11/13/2020, 5:52:57 PM - info - InstanceLoader: BluetoothModule dependencies initialized
> 11/13/2020, 5:52:57 PM - info - InstanceLoader: ScheduleModule dependencies initialized
> 11/13/2020, 5:52:57 PM - info - InstanceLoader: EntitiesModule dependencies initialized
> 11/13/2020, 5:52:57 PM - info - InstanceLoader: HomeAssistantModule dependencies initialized
> 11/13/2020, 5:52:57 PM - info - InstanceLoader: BluetoothLowEnergyModule dependencies initialized
> 11/13/2020, 5:52:57 PM - info - InstanceLoader: StatusModule dependencies initialized
> 11/13/2020, 5:53:00 PM - info - RoutesResolver: EntitiesController {/entities}:
> 11/13/2020, 5:53:00 PM - info - RouterExplorer: Mapped {/entities, GET} route
> 11/13/2020, 5:53:00 PM - info - RoutesResolver: StatusController {/status}:
> 11/13/2020, 5:53:00 PM - info - RouterExplorer: Mapped {/status, GET} route
> 11/13/2020, 5:53:00 PM - warn - BluetoothLowEnergyService: The whitelist and blacklist are empty, no sensors will be created! Please add some of the discovered IDs below to your configuration.
> 11/13/2020, 5:53:03 PM - info - HomeAssistantService: Successfully connected to MQTT broker at mqtt://192.168.86.54:1883
> 11/13/2020, 5:53:03 PM - info - ConfigService: Loading configuration from /opt/nodejs/lib/node_modules/room-assistant/dist/config/definitions/default.js, config/local.yml
> 11/13/2020, 5:53:03 PM - info - ClusterService: Starting mDNS advertisements and discovery
> 11/13/2020, 5:53:03 PM - info - NestApplication: Nest application successfully started
> /opt/nodejs/lib/node_modules/room-assistant/node_modules/@abandonware/noble/lib/hci-socket/hci.js:100
>     this._deviceId = this._socket.bindRaw(deviceId);
>                                   ^
> 
> Error: ENODEV, No such device
>     at Hci.init (/opt/nodejs/lib/node_modules/room-assistant/node_modules/@abandonware/noble/lib/hci-socket/hci.js:100:35)
>     at NobleBindings.init (/opt/nodejs/lib/node_modules/room-assistant/node_modules/@abandonware/noble/lib/hci-socket/bindings.js:78:13)
>     at process.nextTick (/opt/nodejs/lib/node_modules/room-assistant/node_modules/@abandonware/noble/lib/noble.js:60:24)
>     at process._tickCallback (internal/process/next_tick.js:61:11)
> 

I have the following local.yml:
```
global:
  integrations:
    - homeAssistant
    - bluetoothLowEnergy
homeAssistant:
  mqttUrl: 'mqtt://192.168.86.54:1883'
  mqttOptions:
    username: mqtt_honey
    password: D8"&6z`BzdnDUj[x
bluetoothLowEnergy:
  hciDeviceId: 1

```

Am i doing something incorrectly?
Would really appreciate any help..

Thanks,
Alex

**Comment by JesusRo** [comment]
Hey @alexbell62,
Notice that the hciDeviceId param is an array, as well as, on a pi Zero, the first device (and the only one) should be Id 0.
try this:
```
bluetoothLowEnergy:
  hciDeviceId:
    - 0
```

I bought a pi Zero when saw this awesome project and revived some old pi 3 I already had.
It has been working flawless for a week or so. Ordering 6 more like you ;)

**Reply by mKeRix** [reply]
This parameter was actually intended to be a single number, so there might be a bug somewhere in the code. I'll check that out for one of the next versions. Until then: thanks for posting your workaround! :)

**Comment by alexbell62** [comment]
thanks @JesusRo !  Great to hear it is working fine for you (hope im gonna get mine working soon as well)
I did the change but now get different error. Do you know what this can be caused by?

```
pi@bedroom:~/room-assistant $ room-assistant
*** WARNING *** The program 'node' uses the Apple Bonjour compatibility layer of Avahi.
*** WARNING *** Please fix your application to use the native API of Avahi!
*** WARNING *** For more information see <http://0pointer.de/blog/projects/avahi-compat.html>
*** WARNING *** The program 'node' called 'DNSServiceRegister()' which is not supported (or only supported partially) in the Apple Bonjour compatibility layer of Avahi.
*** WARNING *** Please fix your application to use the native API of Avahi!
*** WARNING *** For more information see <http://0pointer.de/blog/projects/avahi-compat.html>
11/13/2020, 6:07:29 PM - info - IntegrationsModule: Loading integrations: home-assistant, bluetooth-low-energy
11/13/2020, 6:07:34 PM - info - NestFactory: Starting Nest application...
11/13/2020, 6:07:34 PM - info - InstanceLoader: AppModule dependencies initialized
11/13/2020, 6:07:34 PM - info - InstanceLoader: ConfigModule dependencies initialized
11/13/2020, 6:07:34 PM - info - InstanceLoader: NestEmitterModule dependencies initialized
11/13/2020, 6:07:34 PM - info - InstanceLoader: IntegrationsModule dependencies initialized
11/13/2020, 6:07:35 PM - info - InstanceLoader: HttpModule dependencies initialized
11/13/2020, 6:07:35 PM - info - InstanceLoader: DiscoveryModule dependencies initialized
11/13/2020, 6:07:35 PM - info - InstanceLoader: ClusterModule dependencies initialized
11/13/2020, 6:07:35 PM - info - InstanceLoader: TerminusModule dependencies initialized
11/13/2020, 6:07:35 PM - info - InstanceLoader: BluetoothModule dependencies initialized
11/13/2020, 6:07:35 PM - info - InstanceLoader: ScheduleModule dependencies initialized
11/13/2020, 6:07:35 PM - info - InstanceLoader: EntitiesModule dependencies initialized
11/13/2020, 6:07:35 PM - info - InstanceLoader: HomeAssistantModule dependencies initialized
11/13/2020, 6:07:35 PM - info - InstanceLoader: BluetoothLowEnergyModule dependencies initialized
11/13/2020, 6:07:35 PM - info - InstanceLoader: StatusModule dependencies initialized
11/13/2020, 6:07:39 PM - info - RoutesResolver: EntitiesController {/entities}:
11/13/2020, 6:07:39 PM - info - RouterExplorer: Mapped {/entities, GET} route
11/13/2020, 6:07:39 PM - info - RoutesResolver: StatusController {/status}:
11/13/2020, 6:07:39 PM - info - RouterExplorer: Mapped {/status, GET} route
11/13/2020, 6:07:39 PM - warn - BluetoothLowEnergyService: The whitelist and blacklist are empty, no sensors will be created! Please add some of the discovered IDs below to your configuration.
events.js:174
      throw er; // Unhandled 'error' event
      ^

Error: bind EADDRNOTAVAIL 192.168.86.24:6425
    at state.handle.lookup (dgram.js:242:18)
    at process._tickCallback (internal/process/next_tick.js:63:19)
    at Function.Module.runMain (internal/modules/cjs/loader.js:834:11)
    at startup (internal/bootstrap/node.js:283:19)
    at bootstrapNodeJSCore (internal/bootstrap/node.js:623:3)
Emitted 'error' event at:
    at state.handle.lookup (dgram.js:243:14)
    at process._tickCallback (internal/process/next_tick.js:63:19)
    [... lines matching original stack trace ...]
    at bootstrapNodeJSCore (internal/bootstrap/node.js:623:3)
```

**Comment by alexbell62** [comment]
actually it may fixed it!! Just one sec checking it now :)

**Comment by JesusRo** [comment]
Error: bind EADDRNOTAVAIL 192.168.86.24:6425 means it couldn´t bind (attach to listen on the) port 6425 I beleive
If I remember correctly, for the pi Zero, I set the user to root for the systemd service.
If followed the instructions in the same way I did, edit your `/etc/systemd/system/room-assistant.service` file and set `User:root`
This is mine for reference:
```
[Unit]
Description=room-assistant service

[Service]
ExecStart=/usr/local/bin/room-assistant
WorkingDirectory=/home/pi/room-assistant
Restart=always
RestartSec=10
User=root

[Install]
WantedBy=multi-user.target
```
Reload the service `sudo systemctl daemon-reload` and restart it `sudo systemctl restart room-assistant`

Give it a try

EDIT: didn't saw previous comment before this sorry xD

**Comment by alexbell62** [comment]
That's great, it all works like magic now. Thanks so much for your help @JesusRo !
